common/js/src/logging/crash-loop-detection.js

```diff
@@ -0,0 +1,132 @@
+/** @license
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Functionality for detecting crash loops. Intended to be an
+ * implementation detail of GoogleSmartCard.Logging.fail() and similar
+ * functions.
+ */
+
+goog.provide('GoogleSmartCard.Logging.CrashLoopDetection');
+
+goog.scope(function() {
+
+const GSC = GoogleSmartCard;
+
+// Name of the chrome.storage key that is used to track the recent crash
+// timestamps. The value is an array of timestamps (as integer milliseconds).
+const STORAGE_KEY = 'crash_timestamps';
+
+// The time window, during which the crashes are counted and compared against
+// |CRASH_LOOP_THRESHOLD_COUNT|.
+const CRASH_LOOP_WINDOW_MILLISECONDS = 60 * 1000;
+// If the number of crashes reaches this constant within the
+// |CRASH_LOOP_WINDOW_MILLISECONDS| time window, then it's considered to be a
+// crash loop.
+//
+// Note that these two constants have to be chosen to be stricter than the
```

Comment:
Maybe I'm confusing something, but I think that a longer time window is stricter than a shorter one? I.e., the sequence of crashes "t=0, t=12, t=24, t=36, t=48" is a violation in the "window=60" configuration, but not in the "window=10" one.

P.S. Replaced "two constants" with their names, to avoid confusion.